tests/unit/amo/reducers/test_reviews.js

```diff
@@ -296,43 +296,124 @@ describe(__filename, () => {
+
+      state = reviewsReducer(
+        state,
+        setUserReviews({
+          pageSize: DEFAULT_API_PAGE_SIZE,
+          reviews: [review],
+          reviewCount: 1,
+          userId,
+        }),
+      );
+      expect(state.byUserId[userId].reviews).toEqual([reviewId]);
+
+      state = reviewsReducer(
+        state,
+        setGroupedRatings({
+          addonId,
+          grouping,
+        }),
+      );
+      expect(state.groupedRatings[addonId]).toEqual(grouping);
+
+      return state;
+    };
+
+    it('clears cached review data', () => {
+      const reviewId = 111;
+      const addonId = 222;
       const addonSlug = 'some-slug';
+      const userId = 333;
+      const grouping = { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 };
 
-      const action = setAddonReviews({
+      let state = loadReviewDataIntoState({
+        addonId,
         addonSlug,
-        pageSize: DEFAULT_API_PAGE_SIZE,
-        reviews: [fakeReview],
-        reviewCount: 1,
+        grouping,
+        reviewId,
+        userId,
       });
 
-      const state = reviewsReducer(undefined, action);
-      const newState = reviewsReducer(state, clearAddonReviews({ addonSlug }));
+      // Clear all data based on a reviewId.
+      state = reviewsReducer(state, clearAddonReviews({ reviewId }));
 
-      expect(newState.byAddon[addonSlug]).toBeUndefined();
+      expect(state.byId[reviewId]).toEqual(undefined);
+      expect(state.byAddon[addonSlug]).toEqual(undefined);
+      expect(state.byUserId[userId]).toEqual(undefined);
+      expect(state.groupedRatings[addonId]).toEqual(undefined);
     });
 
-    it('preserves unrelated add-on reviews', () => {
-      const firstSlug = 'first';
-      const secondSlug = 'second';
+    it('preserves unrelated reviews data', () => {
+      const reviewId = 111;
+      const addonId = 222;
+      const addonSlug = 'some-slug';
+      const userId = 333;
+      const grouping = { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 };
 
-      const review = { ...fakeReview, id: 1 };
-      const action = setAddonReviews({
-        addonSlug: firstSlug,
-        pageSize: DEFAULT_API_PAGE_SIZE,
-        reviews: [review],
-        reviewCount: 1,
+      let state = loadReviewDataIntoState({
+        addonId,
+        addonSlug,
+        grouping,
+        reviewId,
+        userId,
       });
 
-      const state = reviewsReducer(undefined, action);
-      const newState = reviewsReducer(
-        state,
-        clearAddonReviews({ addonSlug: secondSlug }),
-      );
+      // Load some unrelated data.
+      const reviewId2 = 555;
+      const addonId2 = 666;
+      const addonSlug2 = 'some-slug-2';
+      const userId2 = 777;
+
+      state = loadReviewDataIntoState({
+        startState: state,
+        addonId: addonId2,
+        addonSlug: addonSlug2,
+        grouping,
+        reviewId: reviewId2,
+        userId: userId2,
+      });
+
+      state = reviewsReducer(state, clearAddonReviews({ reviewId }));
```

Comment:
Adding expectations before this call to define preconditions would make this test case more readable because one would directly see the impact of `clearAddonReviews`